LUCENE-9997 Revisit smoketester for 9.0 build

janhoy opened the pull request:
https://issues.apache.org/jira/browse/LUCENE-9997

This PR mostly removes solr specific tests, but also
* Fix some python formatting, such as two LF before methods
* Initialize non-initialized variables

janhoy committed:
LUCENE-9997 Revisit smoketester for 9.0 build

janhoy committed:
Remove checkBrokenLinks

janhoy committed:
Add back checkBrokenLinks

cpoerschke commented on dev-tools/scripts/smokeTestRelease.py:
```diff
@@ -658,97 +616,41 @@ def verifyUnpacked(java, project, artifact, unpackPath, gitRevision, version, te
-  solrNotice = open('%s/NOTICE.txt' % unpackPath, encoding='UTF-8').read()
   luceneNotice = open('%s/lucene/NOTICE.txt' % unpackPath, encoding='UTF-8').read()
 
   expected = """
```
Looks like this is no longer used now then, but the `testNotice` function itself still confirms that the `lucene/NOTICE.txt` is readable i.e. the local variable here could be removed (or put to use) but the function itself can stay (assuming no other tests do equivalent checks).

cpoerschke commented on dev-tools/scripts/smokeTestRelease.py:
```diff
@@ -1376,7 +1144,7 @@ def main():
 
   scriptVersion = scriptutil.find_current_version()
-  if not c.version.startswith(scriptVersion + '.'):
+  if not c.version.startswith(scriptVersion + '.') and c.version != scriptVersion:
```
Could this be just this?

```suggestion
  if c.version != scriptVersion:
```

cpoerschke commented on dev-tools/scripts/smokeTestRelease.py:
```diff
@@ -48,6 +48,7 @@
 cygwin = platform.system().lower().startswith('cygwin')
 cygwinWindowsRoot = os.popen('cygpath -w /').read().strip().replace('\\','/') if cygwin else ''
 
+
```
Line 44 above has an `and /solr` that can be removed.

cpoerschke reviewed (commented):
Have carefully reviewed the changes - yes.

Have smoke test run the smoketester (pun intended) - no.

janhoy committed:
Merge branch 'main' into lucene9997-smoketester

janhoy committed:
Review feedback. Remove traces of solr-specific testNotice() method
Move backCompat test up to other "if isSrc" block

janhoy committed:
Review feedback. Bring back the 'checkMaven()' method, as it checks lucene maven artifacts.
But since we dont have pom template files anymore, no need to compare with templates

janhoy committed:
Review feedback. Fix script compatibility by comparing against X.Y instead of X.Y.Z

janhoy committed:
Review feedback. Remove unnecessary if lucene test
Convert some ant commands to gradle

janhoy committed:
Update MANIFEST tests to match the gradle-produced manifest

janhoy committed:
LUCENE-10107 Read multi-line commit from Manifest
Backport from branch_8x

janhoy committed:
Collapse for project in 'lucene' loops and methods taking 'project' as argument
Disable checkJavadocLinks, as this dependency no longer exists in 'scripts' folder

janhoy committed:
Merge branch 'main' into lucene9997-smoketester

janhoy committed:
Review feedback - fix more ant stuff, convert to gradle equivalent

janhoy committed:
Review feedback: Refactor file open

janhoy committed:
Comment out javadoc generation - was only used to check broken links?

janhoy committed:
Fix charset of gpg console output to always be utf-8
Fix two more places to use with open()

janhoy commented:
I have started validating the smoke tester on the artifacts produced by the new build targets in #372 
I find bugs in the gradle build, but also bugs in the smoketester :) 
Once all lights are green, this should be ready to merge

janhoy commented:
> Please ping me if you do!

Ping https://github.com/apache/lucene/pull/372#discussion_r727608870

janhoy committed:
Accept 'LICENSE' without txt or md suffix in top-level

janhoy committed:
Disable vector dictionary abuse exception if started with -Dsmoketester

janhoy committed:
Reformat code

janhoy committed:
Use -Dsmoketester flag when invoking IndexFiles
